scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -580,17 +586,125 @@ class CRIEnumerator( ContainerEnumerator ):
+                add_container = True
+                if glob_list:
+                    add_container = False
+                    for glob in glob_list:
+                        if fnmatch.fnmatch( cname, glob ):
+                            add_container = True
+                            break
+
+                if not add_container:
+                    global_log.log( scalyr_logging.DEBUG_LEVEL_2, "Excluding container '%s' based on glob_list" % short_cid)
+                    continue
+
+                # build a k8s_info structure similar to the one built by _get_containers
+                k8s_info = {}
+                k8s_info['pod_name'] = pod_name
+                k8s_info['pod_namespace'] = pod_namespace
+                k8s_info['k8s_container_name'] = cname
+
+                # get pod and deployment/controller information for the container
+                if k8s_cache:
+                    pod = k8s_cache.pod( pod_namespace, pod_name, current_time )
+                    if pod:
+                        # check to see if we should exclude this container
+                        default_exclude = not k8s_include_by_default
+                        exclude = pod.exclude_pod( container_name=cname, default=default_exclude)
+
+                        # exclude if necessary
+                        if exclude:
+                            if pod.annotations:
+                                global_log.log( scalyr_logging.DEBUG_LEVEL_2, "Excluding container '%s' based on pod annotations, %s" % (short_cid, pod.annotations) )
+                            continue
+
+                        # add a debug message if containers are excluded by default but this container is included
+                        if default_exclude and not exclude:
+                            global_log.log( scalyr_logging.DEBUG_LEVEL_2, "Including container '%s' based on pod annotations, %s" % (short_cid, pod.annotations) )
+
+                        k8s_info['pod_info'] = pod
+                        k8s_info['pod_uid'] = pod.uid
+
+                # build the path to the log file on disk
+                log_path = "%s/%s_%s_%s-%s.log" % (self._log_base, pod_name, pod_namespace, cname, cid)
+
+                # add this container to the list of results
+                result[cid] = {'name': cname, 'log_path': log_path, 'k8s_info': k8s_info }
+        except Exception, e:
+            global_log.error("Error querying containers %s - %s" % (str(e), traceback.format_exc()), limit_once_per_x_secs=300,
+                         limit_key='query-cri-containers' )
+        return result
+
+    def _get_containers_from_filesystem( self, k8s_namespaces_to_exclude ):
+
+        result = []
+
+        # iterate over all files that match our container glob
+        for filename in glob.iglob( self._container_glob ):
+
+            # ignore any files that don't match the format we are expecting
+            m = self._info_re.match( filename )
+            if m is None:
```

Comment:
Not here because it's DEBUG_LEVEL_2, and anything DEBUG_LEVEL_1 or above goes in a separate log file that does not get uploaded to Scalyr servers, and in the event we need to turn on debugging at this level we'll probably want to see all debug messages without limits.

For reference, anything with `error`, `warn`, `info` or `DEBUG_LEVEL_0` will probably need a limit because those things all go to the main agent.log.